modules/axp/Animal/t/RaceHorse.t

```diff
@@ -15,67 +15,92 @@ BEGIN {
 ok( $gg->DOES( "RaceHorse" ), 'RaceHorse object created' );
-is( $gg->{wins}, 0, 'Wins initialised to zero' );
-is( $gg->{places}, 0, 'Places initialised to zero' );
-is( $gg->{shows}, 0, 'Shows initialised to zero' );
-is( $gg->{losses}, 0, 'Losses initialised to zero' );
-like( exception { $gg->named( 'Mr. Ed' ) }, qr/^Static constructor used as instance call/, 'named is static' );
 
-# test 'won'
+my $roWarning = "Cannot assign a value to a read-only accessor at reader";
+
+# test 'wins'/'won'
+is( $gg->wins, 0, 'Wins initialised to zero' );
+like( exception { $gg->wins( 6 ) }, qr/^$roWarning RaceHorse::wins/, "wins is read-only" );
 ok( $gg->won, 'Horse won a race' );
-is( $gg->{wins}, 1, 'Wins incremented' );
+is( $gg->wins, 1, 'Wins incremented' );
 like( exception { RaceHorse->won }, qr/^Instance variable needed/, "won can't be called statically" );
+like( exception { RaceHorse->wins }, qr/^Instance variable needed/, "wins can't be called statically" );
 
-# test 'placed'
+# test 'places'/'placed'
+is( $gg->places, 0, 'Places initialised to zero' );
+like( exception { $gg->places( 6 ) }, qr/^$roWarning RaceHorse::places/, "places is read-only" );
 ok( $gg->placed, 'Horse placed in a race' );
-is( $gg->{places}, 1, 'Places incremented' );
+is( $gg->places, 1, 'Places incremented' );
 like( exception { RaceHorse->placed }, qr/^Instance variable needed/, "placed can't be called statically" );
+like( exception { RaceHorse->places }, qr/^Instance variable needed/, "places can't be called statically" );
 
-# test 'showed '
+# test 'shows'/'showed'
+is( $gg->shows, 0, 'Shows initialised to zero' );
+like( exception { $gg->shows( 6 ) }, qr/^$roWarning RaceHorse::shows/, "shows is read-only" );
 ok( $gg->showed, 'Horse showed in a race' );
-is( $gg->{shows}, 1, 'Shows incremented' );
+is( $gg->shows, 1, 'Shows incremented' );
 like( exception { RaceHorse->showed }, qr/^Instance variable needed/, "showed can't be called statically" );
+like( exception { RaceHorse->shows }, qr/^Instance variable needed/, "shows can't be called statically" );
 
-# test 'lost'
+# test 'losses'/'lost'
+is( $gg->losses, 0, 'Losses initialised to zero' );
+like( exception { $gg->losses( 6 ) }, qr/^$roWarning RaceHorse::losses/, "losses is read-only" );
 ok( $gg->lost, 'Horse lost a race' );
-is( $gg->{losses}, 1, 'Losses incremented' );
+is( $gg->losses, 1, 'Losses incremented' );
 like( exception { RaceHorse->lost }, qr/^Instance variable needed/, "lost can't be called statically" );
 
-# test 'standings'
-( $gg->{wins}, $gg->{places}, $gg->{shows} ) = ( 4, 3, 2 );
+# test constructor validation
+foreach my $field ( qw( wins places shows losses ) ) {
+	foreach my $value ( 'y', 6.6, undef ) {
+		like( exception { RaceHorse->new( name => 'Nonsequituriat', $field => $value ) },
+			qr/^Attribute \($field\) does not pass the type constraint because: Validation failed for 'Int'/,
+			"Non-integer value for $field" );
+	}
+}
```

Comment:
Validation test, since wins/places/shows/losses can now be passed in in the `new` constructor.